include/RAJA/policy/PolicyBase.hpp

```diff
@@ -69,16 +69,26 @@ enum class Pattern {
   taskgraph,
 };
 
+enum class Platform {
+  undefined = 0,
+  host = 1,
+  cuda = 2,
+  omp_target = 4
+};
+
 struct PolicyBase {
 };
 
 template <Policy P = Policy::undefined,
           Launch L = Launch::undefined,
-          Pattern Pat = Pattern::undefined>
+          Pattern Pat = Pattern::undefined,
+          Platform Plat = Platform::host
```

Comment:
This seems iffy to me in a broader sense. omp_parallel_for_exec might want to default to HBM on a KNL, or it might depend on the problem in question. I'd highly recommend testing this on an application on KNL, and working out of HBM on it, while still hitting main memory on a Xeon (non-Phi) system. 

I also know schemas are out, but if an app has to "pick exactly one memory space" for a policy, we don't work on KNL for a lot of applications.

Answer might be "that's fine," but it's worth thinking about